simba/report.py

```diff
@@ -267,6 +291,268 @@ def generate(schedule, scenario, args):
+    :type schedule: eBus-Toolbox.Schedule
+    """
+
+    step = 50
+    # get bin_number
+    max_con = int(max([schedule.rotations[rot].consumption for rot in schedule.rotations]))
+    if max_con % step < step/2:
+        bin_number = ((max_con // step) * step)
+    else:
+        bin_number = (max_con // step) * step + step
+    labels = [f"{i - step} - {i}" for i in range(step, bin_number+step, step) if i > 0]
+    bins = {v_types: [0 for _ in range(bin_number // step)] for v_types in schedule.vehicle_types}
+
+    # fill bins with rotations
+    for rot in schedule.rotations:
+        for v_type in schedule.vehicle_types:
+            if schedule.rotations[rot].vehicle_type == v_type:
+                position = int(schedule.rotations[rot].consumption // step)
+                if position >= bin_number/step:
+                    position -= bin_number // step - 1
+                bins[v_type][position] += 1     # index out of range
+                break
+    # plot
+    fig, ax = plt.subplots()
+    bar_bottom = [0 for _ in range(bin_number//step)]
+    for v_type in schedule.vehicle_types:
+        ax.bar(labels, bins[v_type], width=0.9, label=v_type, bottom=bar_bottom)
+        for i in range(bin_number//step):
+            bar_bottom[i] += bins[v_type][i]
+    ax.set_xlabel('Energieverbrauch in kWh')
+    ax.set_ylabel('Anzahl der Umläufe')
+    ax.set_title('Verteilung der Bustypen über den Energieverbrauch der Umläufe')
+    ax.legend()
+    fig.autofmt_xdate()
+    ax.yaxis.grid(True)
+    plt.savefig(extended_plots_path / "distribution_bustypes_consumption_rotations")
+    plt.close()
+
+
+def charge_type_proportion(extended_plots_path, scenario, schedule):
+    """Plots the absolute number of rotations distributed by charging types on a bar chart.
+
+    :param extended_plots_path: Path argument which combines args.results_directory
+                                and string 'extended_plots'
+    :type extended_plots_path: PosixPath
+    :param scenario: Scenario for with to generate timeseries.
+    :type scenario: spice_ev.Scenario
+    :param schedule: Driving schedule for the simulation. schedule.rotations are used
+    :type schedule: eBus-Toolbox.Schedule
+    """
+    # get plotting data
+    charging_types = {'oppb': 0, 'oppb_neg': 0, 'depb': 0, 'depb_neg': 0}
+    negative_rotations = schedule.get_negative_rotations(scenario)
+    for rot in schedule.rotations:
+        if schedule.rotations[rot].charging_type == 'oppb':
+            if rot in negative_rotations:
+                charging_types['oppb_neg'] += 1
+            else:
+                charging_types['oppb'] += 1
+        elif schedule.rotations[rot].charging_type == 'depb':
+            if rot in negative_rotations:
+                charging_types['depb_neg'] += 1
+            else:
+                charging_types['depb'] += 1
+        else:
+            print("Unknown charging type: ", schedule.rotations[rot].charging_type)
```

Comment:
Don't use print, but warning module. Message could also be extended by rotation ID